src/components/Layout/Header/ActionCenter/components/Details/SwapDetails.tsx

```diff
@@ -1,26 +1,64 @@
 import { Button, ButtonGroup, Link, Progress, Stack } from '@chakra-ui/react'
-import React from 'react'
+import type { Swap } from '@shapeshiftoss/swapper'
+import { SwapStatus } from '@shapeshiftoss/swapper'
+import React, { useMemo } from 'react'
 import { useTranslate } from 'react-polyglot'
 
+import { useStreamingProgress } from '@/components/MultiHopTrade/components/TradeConfirm/hooks/useStreamingProgress'
 import { Row } from '@/components/Row/Row'
 import { RawText } from '@/components/Text'
+import { bnOrZero } from '@/lib/bignumber/bignumber'
 
 type SwapDetailsProps = {
-  isStreaming?: boolean
   txLink?: string
+  swap: Swap
 }
 
-export const SwapDetails: React.FC<SwapDetailsProps> = ({ isStreaming, txLink }) => {
+export const SwapDetails: React.FC<SwapDetailsProps> = ({ txLink, swap }) => {
   const translate = useTranslate()
+  const { isStreaming } = swap
+
+  const streamingProgress = useStreamingProgress({ swap })
+
+  const {
+    numSuccessfulSwaps: thorchainNumSuccessfulSwaps,
+    totalSwapCount: thorchainTotalSwapCount,
+    isComplete: isThorchainStreamingComplete,
+  } = streamingProgress ?? {}
+
+  const isComplete = isThorchainStreamingComplete || swap.status === SwapStatus.Success
```

Comment:
q: do we want to mix and mash this in `<SwapDetails />`? Or can we perhaps have a streaming variant that would consume streaming hook exclusively, not to add overhead here (and so we wouldn't even render the streaming tree if it's not a streaming swap, meaning we could never end up in polling streaming for regular swaps)

Reply:
I can, it doesn't make so much sense now that streaming polls are not polling if the swap is not a streaming one, but it doesn't hurt

Done in [7ccd82e](https://github.com/shapeshift/web/pull/9670/commits/7ccd82ebc7a60c732a55ecf33e337434aaa655cb)